Sming/SmingCore/Network/TcpClient.cpp

```diff
@@ -53,6 +53,14 @@ TcpClient::~TcpClient()
 	}
 }
 
+bool TcpClient::connect(const char * server, int port, boolean useSsl /* = false */, uint32_t sslOptions /* = 0 */)
+{
+	if (isProcessing()) return false;
+
+	state = eTCS_Connecting;
+	return TcpConnection::connect(server, port, useSsl, sslOptions);
```

Comment:
I don't know the reason existing code passes cstr. It does not make sense to me but again, I'm not sure. I think it should be internally only 2 methods, one that gets hostname and another that takes IP address, and again, internally the one that takes hostname should do DNS lookup and call the IP one. To keep existing class interface there should be one more method that takes String object and calls the cstr version. I've ran couple of examples that use client class and did not observe any instabilities but this was limited test.

Anyway, this may be worth cleaning up to have concise and consistent class interface.